src/v/cloud_storage/tests/remote_partition_test.cc

```diff
@@ -2417,9 +2417,17 @@ FIXTURE_TEST(test_out_of_range_spillover_query, cloud_storage_fixture) {
           return what.find("Failed to query spillover manifests") != what.npos;
       });
 
-    // BUG: This assertion is disabled because it currently fails.
-    // test_log.debug("Timestamp undershoots the partition");
-    // BOOST_TEST_REQUIRE(timequery(*this, model::timestamp(100), 3 * 6));
+    // Timestamp undershoots the partition.
+    // It shouldn't throw. But as an interim workaround we throw until
```

Comment:
Is this tracked anywhere as a bug? May be worth linking an issue if there is one

Reply:
Actually the bug is more trickier than this comment captures. I have added some additional tests. They should make it much more clearer now what the bug actually is https://github.com/redpanda-data/redpanda/pull/18097/commits/f52c3e9be3ecfb984acd8f2d98beeae70e81c390

I'm trying to avoid to make changes for that in this PR. We can fix it in subsequent work. I'm afraid that change in behavior will affect too many tests and I'll never get to merge the fix for higher severity bugs which this PR already covers.